nemo_skills/pipeline/openrlhf/ppo.py

```diff
@@ -0,0 +1,446 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import List
+
+import os
+import nemo_run as run
+import typer
+from datetime import datetime
+from dataclasses import dataclass
+from omegaconf import OmegaConf
+from typing import Optional
+
+from nemo_skills.pipeline.openrlhf import openrlhf_app
+from nemo_skills.pipeline import add_task, check_if_mounted, get_cluster_config, run_exp
+from nemo_skills.pipeline.app import app, typer_unpacker
+from nemo_skills.pipeline.generate import wrap_cmd
+from nemo_skills.utils import setup_logging
+
+LOG = logging.getLogger(__file__)
+
+@dataclass
+class PPOOpenRLHFTask:
+    model: str
+    reward_model: str
+    output_dir: str
+    prompt_data: str
+    num_gpus: int
+    num_nodes: int
+    expname: str
+    disable_wandb: bool
+    wandb_project: str
+    timeout: str
+    extra_arguments: str = ""
+    logging_params: str = ""
+
+
+    def get_ray_server_cmd(self, start_cmd, cluster_config):
```

Comment:
can you please move this and the next functions to utils, and reuse from there, so that we can keep this logic in one place?